### ISSUE TITLE
Special characters ' in translated country url issue (iso 3166-1)

### ISSUE THREAD

**Issue by seve26** [CinemaPress/CinemaPress#19]
Hello,
Cinemapress is really amazing, everything works great except for one little thing... The language of my site is set to French. I have a problem with the path of the url automatically generated with the translation (iso 3166-1) for the country: USA. In my language, `The United States Of America` is translated with iso 3166-1 as `États-Unis d'Amérique`. The system therefore generates this url: `https://example.com/country-États-Unis-d'Amérique` however when I go to the url, there is no movie, nothing in the category.

I think it's a problem with the apostrophe `'` in the url.

Is it possible to manually modify the path of the translated url by replacing `country-États-Unis-d'Amérique` with `country-USA`?

Is it possible to add a condition when the path of the generated and translated url is created that will replace some special characters with other characters? Only in the url path, not in the title of the page created.

For example:

- transform all `'` into `-`
- transform all `é` into `e`
- transform all `è` into `e`
- transform all `ê` into `e`
- transform all `à` into `a`
- transform all `ç` into `c`
- transform all `ù` into `u`
- transform all `û` into `u`
- transform all uppercase letters to lowercase letters

So basically transform this: `https://example.com/country-États-Unis-d'Amérique` to this:` https://example.com/country-etats-unis-d-amerique`

Best Regards

**Comment by zeldaroot** [comment]
Hello. Thank you for using CinemaPress.

![image](https://user-images.githubusercontent.com/56031696/136485100-1be059a0-3be0-4059-8717-e4e2e0074a48.png)

1. Change to a custom value `URL settings`;
2. This setting will cause the URL to look like this `/country-c-6tats-Unis-d_Am6rique`
3. Reversible transform `'` <-> `_` | `è` <-> `3` | `é` <-> `6`. The dash `-` cannot be used, as it is separator.
4. This setting will cause the URL to look like this `/country-c-6tats-unis-d_am6rique`

I think you can do without transformation. The search for movies goes without errors.
 `/country-c-Étatstats-Unis-d'Amérique`
or
 `/country-c-états-unis-d'amérique`

Update the engine:
`cinemapress upd`
`cinemapress backup`
`cinemapress update`

If you have any questions, please ask.

**Comment by zeldaroot** [comment]
Hello,
Do you see your backups on Mega?

![image](https://user-images.githubusercontent.com/56031696/136638997-bcd84eda-cc33-4a30-9dfc-1deb933815f8.png)

![image](https://user-images.githubusercontent.com/56031696/136639083-c270d536-e49f-468a-90d1-6fed26fc4125.png)

New options after update page (F5).
Please refresh the page, but first you must update the engine without errors.

In case of errors, you can always restore the website from the last successful backup. It is located in the `latest` folder MEGA. If you need to restore a website for a different date, rename the folder to `latest`.

If you have problems with the backup, go to MEGA https://mega.nz/fm/account/security and end all sessions. Sometimes MEGA puts restrictions on file transfers.

![image](https://user-images.githubusercontent.com/56031696/136639619-9074763a-c8d3-480c-a78d-406ae8453e03.png)

After that, I managed to successfully restore the website from the backup.

![image](https://user-images.githubusercontent.com/56031696/136639717-f0bb2cf5-33e1-436f-b822-19808e6b7f91.png)

**Comment by seve26** [comment]
Hello,
Thanks a lot for the reply, I restored with a backup that had been backed up 4 days ago because the latest backup contained errors. This time, I didn't have problem updating Cinemapress with `cinemapress update`.

![cinemapress](https://user-images.githubusercontent.com/1296080/136642307-0c0f26ef-de9b-408c-af2b-ddfec4ef0b96.jpg)

 Now I can see the new option `change from` and `change to` once I select `Customary` in Unique movie url setting:

![working](https://user-images.githubusercontent.com/1296080/136642492-49e4f189-5bf0-4156-b8c6-5ac725302bb4.jpg)

Amazing! Thanks again!
Best Regards
